src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -219,4 +232,127 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
+     * a name corresponding to the appium sub module as prefixed in log message
+     * (logger name is transformed to lower case, no spaces and prefixed with
+     * "appium.service.").</li> </br>
+     * Example log-message: "[ADB] Cannot read version codes of " is logged by
+     * logger: {@code appium.service.adb} at level {@code INFO}. </br>
+     * Example log-message: "[debug] [XCUITest] Xcode version set to 'x.y.z' "
+     * is logged by logger {@code appium.service.xcuitest} at level
+     * {@code DEBUG}.
+     * 
+     * </br>
+     * </br>
+     * 
+     * @see #addSlf4jLogMessageConsumer(BiConsumer)
+     */
+    public void enableDefaultSlf4jLoggingOfOutputData() {
+        addSlf4jLogMessageConsumer((logMessage, ctx) -> {
+            if (ctx.level().equals(DEBUG)) {
+                ctx.logger().debug(logMessage);
+            } else {
+                ctx.logger().info(logMessage);
+            }
+        });
+    }
+
+    /**
+     * When a complete log message is available (from server output data) that
+     * message is parsed for its slf4j context (logger name, logger level etc.)
+     * and the specified {@code BiConsumer} is invoked with the log message and
+     * slf4j context.</br>
+     * </br>
+     * Use this method only if you want a behavior that differentiates from the
+     * default behavior as enabled by method
+     * {@link #enableDefaultSlf4jLoggingOfOutputData()}. </br>
+     * </br>
+     * NOTE: You might want to call method {@link #clearOutPutStreams()} before
+     * calling this method. </br>
+     * </br>
+     * implementation detail:
+     * <li>if log message begins with {@code [debug]} then log level is set to
+     * {@code DEBUG}, otherwise log level is {@code INFO}.</li>
+     * <li>the appium sub module name is parsed from the log message and used as
+     * logger name (prefixed with "appium.service.", all lower case, spaces
+     * removed). If no appium sub module is detected then "appium.service" is
+     * used as logger name.</li> </br>
+     * Example log-message: "[ADB] Cannot read version codes of " is logged by
+     * {@code appium.service.adb} at level {@code INFO} </br>
+     * Example log-message: "[debug] [XCUITest] Xcode version set to 'x.y.z' "
+     * is logged by {@code appium.service.xcuitest} at level {@code DEBUG}
+     * 
+     * @param slf4jLogMessageConsumer
+     *            BiConsumer block to be executed when a log message is
+     *            available.
+     */
+    public void addSlf4jLogMessageConsumer(
+            BiConsumer<String, Slf4jLogMessageContext> slf4jLogMessageConsumer) {
+        addLogMessageConsumer(logMessage -> {
+            slf4jLogMessageConsumer.accept(logMessage, parseSlf4jContextFromLogMessage(logMessage));
+        });
+    }
+
+    static Slf4jLogMessageContext parseSlf4jContextFromLogMessage(String logMessage) {
+        Matcher m = LOGGER_CONTEXT_PATTERN.matcher(logMessage);
+        String loggerName = "appium.service";
```

Comment:
This should be in the constant

Reply:
I'm not following you on this. What should be in the constant? Which constant?